### PR TITLE
deploy: stop defaulting to ImageStreamTag unconditionally

### PR DESCRIPTION
**Pull request by kargakis**
@deads2k PTAL

### CODE REVIEW

**Comment by deads2k** [comment]
add a test to `test-cmd` to create the problem images and see success so we'll know if we slip again, then lgtm

**Comment by kargakis** [comment]
[test]

**Comment by kargakis** [comment]
I cloned the template into our repo since I have an open pull that changes those templates to be correct: https://github.com/jboss-openshift/application-templates/pull/190

File: test/cmd/deployments.sh [hunk 1]
```diff
@@ -93,6 +93,11 @@ os::cmd::expect_success_and_text "oc get dc --show-labels" "app=dockerbuild,temp
 os::cmd::expect_success_and_text "oc get dc frontend --show-labels" "app=dockerbuild,template=application-template-dockerbuild"
 os::cmd::expect_success_and_not_text "oc get dc" "app=dockerbuild,template=application-template-dockerbuild"
 os::cmd::expect_success_and_not_text "oc get dc frontend" "app=dockerbuild,template=application-template-dockerbuild"
+
+os::cmd::expect_success "oc process -f https://raw.githubusercontent.com/openshift/origin/master/test/testdata/old-template.json | oc create -f -"
```
Comment (deads2k) [inline test/cmd/deployments.sh]: can use local file now.

**Comment by kargakis** [comment]
Comments addressed, [merge]

**Comment by kargakis** [comment]
yum [merge]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/9989 [merge]

**Comment by kargakis** [comment]
secrets [merge]

**Comment by kargakis** [comment]
appliedclusterresourcequota flake [merge]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/9965 [merge]

**Pull request merged**